unity-renderer/Assets/Scripts/MainScripts/DCL/Components/CameraModeArea/CameraModeArea.cs

```diff
@@ -0,0 +1,176 @@
+        {
+            Dispose();
+        }
+
+        void IEntityComponent.Initialize(IParcelScene scene, IDCLEntity entity)
+        {
+            Initialize(scene, entity, Environment.i.platform.updateEventHandler, DataStore.i.player.playerCollider.Get());
+        }
+
+        internal void OnModelUpdated(in Model newModel)
+        {
+            if (!IsValidCameraMode(newModel.cameraMode))
+            {
+                newModel.cameraMode = CommonScriptableObjects.cameraMode.Get();
+            }
+
+            bool cameraModeChanged = newModel.cameraMode != areaModel.cameraMode;
+            areaModel = newModel;
+
+            if (cameraModeChanged && isPlayerInside)
+            {
+                areasController.ChangeAreaMode(this, areaModel.cameraMode);
+            }
+        }
+
+        internal void Initialize(in IParcelScene scene, in IDCLEntity entity, in IUpdateEventHandler updateEventHandler, in Collider playerCollider)
+        {
+            areaScene = scene;
+            areaEntity = entity;
+            this.updateEventHandler = updateEventHandler;
+            this.playerCollider = playerCollider;
+
+            updateEventHandler.AddListener(IUpdateEventHandler.EventType.Update, Update);
+        }
+
+        internal void Update()
+        {
+            bool playerInside = IsPlayerInsideArea();
+
+            switch (playerInside)
+            {
+                case true when !isPlayerInside:
+                    areasController.AddInsideArea(this);
+                    break;
+                case false when isPlayerInside:
+                    areasController.RemoveInsideArea(this);
+                    break;
+            }
+            isPlayerInside = playerInside;
+        }
+
+        internal void Dispose()
+        {
+            OnAreaDisabled();
+            updateEventHandler.RemoveListener(IUpdateEventHandler.EventType.Update, Update);
+        }
+
+        private bool IsValidCameraMode(in CameraMode.ModeId mode)
+        {
+            if (validCameraModes == -1)
+                return true;
+
+            return ((1 << (int)mode) & validCameraModes) != 0;
+        }
+
+        private bool IsPlayerInsideArea()
+        {
+            if (areaEntity == null || areaScene == null)
+            {
+                return false;
+            }
+
+            if (areaScene.sceneData.id != CommonScriptableObjects.sceneID.Get())
+            {
+                return false;
+            }
+
+            Vector3 center = areaEntity.gameObject.transform.position;
+            Quaternion rotation = areaEntity.gameObject.transform.rotation;
+
+            Collider[] colliders = Physics.OverlapBox(center, areaModel.area.box * 0.5f, rotation,
+                PhysicsLayers.avatarTriggerMask, QueryTriggerInteraction.Collide);
+
+            if (colliders.Length == 0)
+            {
+                return false;
+            }
+
+            return colliders.Any(collider => collider == playerCollider);
```

Comment:
If this is the only `LINQ` usage in the script, can we turn that into a regular `for` structure and remove `LINQ`?